rs/registry/nns_data_provider/src/registry.rs

```diff
@@ -373,14 +392,6 @@ async fn deserialize_and_dechunk_get_value_result(
         ))
     })?;
 
-    // Handle no reply. Not sure how this is possible.
-    let Some(result) = result else {
```

Comment:
Note to reviewer - previous canister_client returned Option<Vec<u8>> instead of just Vec<u8> because it translated a direct RequestStatus field (reply), which was Option<Vec<u8>> but appears to be mutually exclusive with the reject_message field that was Option<String>.  The ic-agent library doesn't return this type, thus a lot of type changes w/r/t to Option in this file's decoding logic.